website/client/js/controllers/userCtrl.js

```diff
@@ -15,6 +15,13 @@ habitrpg.controller("UserCtrl", ['$rootScope', '$scope', '$location', 'User', '$
     $scope.$watch('_editing.profile', function(value){
       if(value === true) $scope.editingProfile = angular.copy(User.user.profile);
     });
+    
+    $scope.isBasicAnimal = function(name, type) {
+      var tmp = name.split('-');
+      $scope[type + "Egg"] = tmp[0];
+      $scope[type + "Potion"] = tmp[1];
+      return Content.hatchingPotions[$scope[type + "Potion"]] ? true : false;
+    }
```

Comment:
Where do you think I should put it such that it is available in both scopes?
On Wed, Jul 20, 2016 at 4:47 PM Kaitlin Hipkin <notifications@github.com>
wrote:

> In website/client/js/controllers/userCtrl.js
> <https://github.com/HabitRPG/habitrpg/pull/7768#discussion_r71612902>:
>
> > @@ -15,6 +15,13 @@ habitrpg.controller("UserCtrl", ['$rootScope', '$scope', '$location', 'User', '$
> >      $scope.$watch('_editing.profile', function(value){
> >        if(value === true) $scope.editingProfile = angular.copy(User.user.profile);
> >      });
> > +
> > +    $scope.isBasicAnimal = function(name, type) {
> > +      var tmp = name.split('-');
> > +      $scope[type + "Egg"] = tmp[0];
> > +      $scope[type + "Potion"] = tmp[1];
> > +      return Content.hatchingPotions[$scope[type + "Potion"]] ? true : false;
> > +    }
>
> Code that does the same thing for the same reason in two different places
> shouldn't be written twice. Write it somewhere else and use it twice
> instead.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/HabitRPG/habitrpg/pull/7768/files/cc4cd30e10d6b6bfc88a951ff604f99ee02a5ac0..0208c75f66098b621c066f18b06e12153cc25893#r71612902>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALz_8xg8FgpVL_7dvf-xIWEW6moASqprks5qXpdagaJpZM4JIjd_>
> .
>